### PR TITLE
github: integrate the GH webhooks directly with the bridge process

### PR DESCRIPTION
**Pull request by samoht**
This needs the dev version of ocaml-github to work. Not fully tested yet.

Signed-off-by: Thomas Gazagnaire <thomas@gazagnaire.org>

### CODE REVIEW

File: bridge/github/main.ml [hunk 1]
```diff
@@ -222,11 +206,16 @@ let public_branch =
...
   Arg.(value & opt string "github-metadata" doc)
 
+let uri =
+  let parse str = `Ok (Uri.of_string str) in
+  let print ppf uri = Fmt.string ppf @@ Uri.to_string uri in
+  parse, print
+
 let webhook =
   let doc =
     Arg.info ~doc:"Pubic URI of the GitHub webook server" ["webhook"]
```
Comment (dsheets) [inline bridge/github/main.ml]: s/Pubic/Public/

**Comment by dsheets** [comment]
LGTM. Thanks for the attribution. It would be most excellent to have the web hook stuff polished up and contributed back upstream to ocaml-github as an optional subpackage (to isolate the extra deps like nocrypto) or put into a new package like ocaml-github-hooks.

**Comment by samoht** [comment]
All green, merging!

**Pull request merged**